runtime/lua/vim/lsp/buf.lua

```diff
@@ -177,15 +177,27 @@ end
 ---     - bufnr (number|nil):
 ---         Restrict formatting to the clients attached to the given buffer, defaults to the current
 ---         buffer (0).
+---
 ---     - filter (function|nil):
----         Predicate to filter clients used for formatting. Receives the list of clients attached
+---         Predicate used to filter clients. Receives a client as argument and must return a
+---         boolean. Clients matching the predicate are included. Example:
+---
+---         <pre>
+---         -- Never request typescript-language-server for formatting
+---         vim.lsp.buf.format {
+---           filter = function(client) return client.name ~= "tsserver" end
+---         }
+---         </pre>
+---
+---     - get_clients (function|nil):
```

Comment:
I've read the docs over and over and I still don't get this. We have `filter` to filter the clients, and `get_clients` to filter the clients. wtf?

Reply:
I'm inclined to remove `get_clients` and just go with changing the signature of `filter`.

The use case mentioned in the original PR that added the format function with the `clients -> clients` filter was inspecting the clients and returning a result based on which clients are available.  E.g if null-ls is present,  use that - otherwise use the other clients. But if one really needed to do that they could also have a predicate that calls get_active_clients itself.

A `client -> bool` signature should be easier to use for most cases and fits better with the name `filter` which typically takes a predicate

Reply:
> I'm inclined to remove `get_clients` and just go with changing the signature of `filter`.

💯